### ISSUE TITLE
[Feature Request] Add containers, types, mimetypes for archives

### ISSUE THREAD

**Issue by labrys** [guessit-io/guessit#272]
.. for e.g. RAR packed files

```
$ guessit "Treme.1x03.Right.Place,.Wrong.Time.HDTV.XviD-NoTV.rar"
For: Treme.1x03.Right.Place,.Wrong.Time.HDTV.XviD-NoTV.rar
GuessIt found: {
    "title": "Treme",
    "season": 1,
    "episode": 3,
    "episode_title": "Right Place, Wrong Time",
    "format": "HDTV",
    "video_codec": "XviD",
    "release_group": "NoTV",
    "container": "rar",
    "mimetype": "application/x-rar-compressed",
    "type": "episodearchive"
}
```

**Comment by Diaoul** [comment]
Not sure type is a good idea here. It should remain `episode` so the main API call is still `guessit(..., type='episode')`

**Comment by Toilal** [comment]
:-1: `type` should not vary with container.

And mimetypes guessing depends on OS mimetype configuration, so we can't rely too much on this.

As an exemple, on my computer mimetype is `application/rar` but `application/x-rar-compressed` in yours.

**Comment by labrys** [comment]
My suggestion for changing the type based on container was a direct reflection of your [documentation](http://guessit.readthedocs.org/en/latest/properties.html#properties):
> ##### type
###### Type of the file.
* unknown, movie, episode, moviesubtitle, episodesubtitle

`episodesubtitles` and `moviesubtitles` are separated based on the container.  I'm perfectly fine not separating it, but it should be consistent across all types then.

As for the mimetype issue, also per your docs.
> Guessed values may vary based on OS native support of mime type.
So I'm not sure the point you're trying to make regarding the mimetypes.  Also rar may not be the best example for an archive in this case as it doesn't have an official IANA defined mimetype, however it is common with video files.

Regardless of how you decide to go with this, I feel detection of archives is an important feature as they are a distinct class and occur frequently with videos.

**Comment by labrys** [comment]
Examples of invalid guesses for .rars (guessit v2.04):
```python
>>> guessit('Back to the future.rar')
MatchesDict([('title', 'Back to the future rar'), ('type', 'movie')])
>>> guessit('Scrubs S01E02 My Mentor.rar')
MatchesDict([('title', 'Scrubs'), ('season', 1), ('episode', 2), ('episode_title', 'My Mentor rar'), ('type', 'episode')])
>>> guessit('Scrubs S01E02 My Mentor 720p BluRay.rar')
MatchesDict([('title', 'Scrubs'), ('season', 1), ('episode', 2), ('episode_title', 'My Mentor'), ('screen_size', '720p'), ('format', 'BluRay'), ('release_group', 'rar'), ('type', 'episode')])
```

**Comment by Toilal** [comment]
I agree @labrys, subtitles should not change the type of the file. There should be a `subtitle` property ... Maybe a `container_type`, with possible values as `subtitle` or `archive` ?

**Comment by labrys** [comment]
I think a `container_type` could be useful, as it could also apply to #273.  One possible issue I can see with detection is containers within containers.  Not the contents of a container, as that would require examining the file but compound file extensions (e.g. `.tar.gz`, '.mkv.rar'), possibly handling this with a list like episodes/seasons/others and the primary being the final extension.

**Comment by Toilal** [comment]
Implemented on `develop`: archive extensions are now recognised — e.g. `…HDTV.XviD-NoTV.rar` yields `container: rar` with `mimetype: application/vnd.rar`, and the rest of the name parses cleanly. The `type` stays `movie`/`episode` (guessit keeps `type` to the media class rather than introducing an `episodearchive` type).

Closing as resolved — please reopen if you specifically need a dedicated archive `type`.
